src/query/frontend/opencypher/grammar/Cypher.g4

```diff
@@ -275,6 +276,8 @@ reduceExpression : accumulator=variable '=' initial=expression ',' idInColl '|'
 
 extractExpression : idInColl '|' expression ;
 
+existsExpression : patternPart ;
```

Comment:
@antaljanosbenjamin do you think we should use here `anonymousPatternPart` as we don't support `( variable '=' anonymousPatternPart ) ` in `exists`? From what I understood there would need to refactor code as there is functionality for PatternPart but not for `anonymousPatternPart`

Reply:
In my opinion, we don't need to do so as it doesn't complicate code for worse, but makes it easier now to deliver, but it is not best practice nevertheless.

Reply:
I would say use only what we need. In my opinion both `anonymousPatternPart` and `patternElement` is suitable here.

What is the reason to use `patternPart` instead? What extra costs does it have?

Reply:
Actually exists doesn't support whole `patternPart`, only `anonymusPatternPart`. Because in `exists`, you can't write the following which is allowed pattern part: `... WHERE exists(p = (n)-[]->(m));`
and in `patternPart` this is allowed: `( variable '=' anonymousPatternPart )`

Reply:
I decided in the end to stick to patternPart as it has better error handling capabilities if the user messes up something with patterns

Reply:
I.E. I can throw a detailed message if the user specifies p=()-[]-() telling him that there is no identifier for pattern in exists. Whereas if I use an anonymous pattern part, I don't handle that case but it's somewhere in the air (atom visiting)